core/src/main/java/org/bouncycastle/math/ec/ECCurve.java

```diff
@@ -797,11 +798,15 @@ public static BigInteger inverse(int m, int[] ks, BigInteger x)
 
         private static FiniteField buildField(int m, int k1, int k2, int k3)
         {
-            if (k1 == 0)
+            if (m > Properties.asInteger("org.bouncycastle.ec.max_f2m_field_size", 1142))  // twice 571
             {
-                throw new IllegalArgumentException("k1 must be > 0");
+                throw new IllegalArgumentException("field size out of range: " + m);
             }
 
+            int[] exponents = (k2 | k3) == 0
+                ? new int[]{ 0, k1, m }
+                : new int[]{ 0, k1, k2, k3, m };
+
```

Comment:
This is incorrect as all it does is creates an unused variable

The section of the code that has been back-ported here was originally introduced in the code base as part of the commit [e078224](https://github.com/lineaje-labs-gos/bc-java/commit/e0782246fa759eb2af21433cfaf1df8f5cc7253b).
In the original commit, the code base was refactored in other files to use "exponents" by using `FiniteFields.getBinaryExtensionField`
If the intent is to use `exponents`, then all the other refactoring done in that commit should also be back ported OR the change done to introduce `exponents` should be removed

Reply:
Removed the unused variable in a20509975142ec935a3b638d962ea017faff6981